app/src/main/res/values/strings.xml

```diff
@@ -3,4 +3,7 @@
     <string name="rationale_camera">This app needs access to your camera so you can take pictures.</string>
     <string name="rationale_location_contacts">This app needs access to your location and contacts to know where and who you are.</string>
     <string name="rationale_sms">This app needs access to your sms to read all your great messages.</string>
+    <string name="rationale_ask_again">This app needs to open permissions before using the feature.</string>
+    <string name="setting">SETTING</string>
```

Comment:
nit: replace these two Strings with 'Settings' and 'Cancel'.  Android will take care of capitalizing them when appropriate.